concrete/src/User/User.php

```diff
@@ -585,19 +588,29 @@ public function getUserAccessEntityObjects()
                 return PermissionAccessEntity::getForUser($this);
             }
 
-            if ($session->has('accessEntities')) {
+            // If a user is logged in and running a script to get the user access entities
+            // return the correct access entities
+            if ($session->has('accessEntities') && $this->getUserID() == $session->get('uID')) {
                 $entities = $session->get('accessEntities');
-            } else {
+            } elseif ($this->getUserID() == $session->get('uID')) {
                 $entities = PermissionAccessEntity::getForUser($this);
                 $session->set('accessEntities', $entities);
                 $session->set('accessEntitiesUpdated', time());
+            } else {
+                $entities = PermissionAccessEntity::getForUser($this);
+
             }
         } else {
-            $group = Group::getByID(GUEST_GROUP_ID);
-            if ($group) {
-                $entities = [GroupEntity::getOrCreate($group)];
+
+            if ((int) $this->getUserID() > 0) {
```

Comment:
What about
```php
            if ($this->isRegistered()) {
```
instead?

Reply:
I mean its the exact same code
https://github.com/concrete5/concrete5/blob/c332a91cee8b16d149bb2d7889c8fec8799d3a7b/concrete/src/User/User.php#L328-L331